bin/bioentities2json.py

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env python
+
+"""bioentities2json
+
+This script generate json output from taking input as biomart species metadata used in Ensembl update for bioentities collection loading in Solr.
+
+running syntax
+
+python bioentities2json.py -i homo_sapiens.ensgene.tsv
+
+i - Input tsv (tab separated values) is Biomart species metadata file consisting of gene ids as rows and associated gene attributes as columns
+
+Converted json will be printed in STD output
+"""
+
+import json
+import argparse
+import pandas as pd
+
+
+def get_args():
+    arg_parser = argparse.ArgumentParser()
+    arg_parser.add_argument('-i', '--input_tsv',
+                            required=True,
+                            help='A TSV file of gene metadata')
+    args = arg_parser.parse_args()
+    return args
+
+def read_tsv_file(tsv_file_path):
+    tsv_file = pd.read_csv(tsv_file_path, sep='\t', header=0)
+    return tsv_file
+
+def join_string(list_string):
+    # Join the string based on '_' delimiter
+    string = '_'.join(list_string)
+    return string
+
+def get_species_name_from_file(file_name):
+    species = join_string(file_name.split("/")[-1].split('_')[0:2]).capitalize()
+    return species
+
+def get_property_weight(property_name):
+    property_names = ['gene_biotype', 'hgnc_symbol', 'uniprot', 'entrezgene']
+
+    if property_name == 'synonym':
```

Comment:
This lines could be all replaced by loading a yaml file that has all this information and gets loaded into a dict. Then it is also easier to change these weights without having to touch the code.

Reply:
Property yaml file contains predefined weights for an attribute that is given priority while searching in webapp